cli/src/command/wallet.rs

```diff
@@ -272,32 +250,28 @@ pub async fn restore_command(storage_path: &Path, snapshot_path: &Path, backup_p
         .await?;
 
     let password = get_password("Stronghold backup password", false)?;
-    wallet.restore_backup(backup_path.into(), password, None, None).await?;
-
-    println_log_info!(
-        "Wallet has been restored from the backup file \"{}\".",
-        backup_path.display()
-    );
-
-    Ok(wallet)
+    if let Err(e) = wallet.restore_backup(backup_path.into(), password, None, None).await {
+        // Clean up a failed restore (typically produces a wallet without a secret manager)
+        // TODO: a better way would be to not create any files/dirs in the first place when it's not clear yet whether
+        // the restore will be successful
+        std::fs::remove_dir_all(storage_path)?;
```

Comment:
If there is still a db with accounts this will delete it, because it will error with `wallet error: backup failed can't restore backup when there are already accounts`
Result is nothing is restored but the db is gone

To reproduce:
1. create an account
2. create a backup
3. try to restore backup -> will fail and delete db

Reply:
well, after that the restore will work :trollface: